### PR TITLE
doc: add checkpoint conversion in tutorials

### PR DESCRIPTION
**Pull request by charlesli640**
# Description

Add/Move checkpoint conversion utils README to tutorials, corrected URL link in sphinx-build warning.
Also aggregate duplicated descriptions of checkpoint conversion in post training tutorial.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: docs/tutorials/posttraining/convert_hf_checkpoint.md
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/convert_hf_checkpoint.md:1]: Can we combine this doc to `convert_checkpoint.md`?

**Comment by SurbhiJainUSC** [comment]
@hengtaoguo - please review.

File: docs/tutorials/posttraining/convert_hf_checkpoint.md
Comment (charlesli640) [reply docs/tutorials/posttraining/convert_hf_checkpoint.md:1]: OK. Let me do that.

File: docs/guides/checkpointing_solutions/convert_checkpoint.md
Comment (hengtaoguo) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md:1]: Thanks! I see you didn't remove the previous [README](https://github.com/AI-Hypercomputer/maxtext/blob/main/src/MaxText/utils/ckpt_conversion/README.md). What's your opinion on this: should we remove the old one, or keep both copies (which may require dual-updates in the future)?

File: docs/guides/checkpointing_solutions/convert_checkpoint.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Checkpoint conversion utilities
+
+This guide provides instructions for using the scripts that convert model checkpoints bidirectionally between Hugging Face and MaxText formats.
```
Comment (hengtaoguo) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md]: Maybe add a hyper-link to the conversion scripts since we moved to a different place?

https://github.com/AI-Hypercomputer/maxtext/tree/main/src/MaxText/utils/ckpt_conversion

File: docs/guides/checkpointing_solutions/convert_checkpoint.md
Comment (charlesli640) [reply docs/guides/checkpointing_solutions/convert_checkpoint.md:1]: We better avoid duplication. I just removed most content in README, and added a hyper-link to docs/guides/...

File: docs/guides/checkpointing_solutions/convert_checkpoint.md (hunk 1 above)
Comment (charlesli640) [reply docs/guides/checkpointing_solutions/convert_checkpoint.md]: Done.

**Review by hengtaoguo — APPROVED** [review]

File: docs/guides/checkpointing_solutions/convert_checkpoint.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+```bash
+# -- Model configuration --
+export HF_MODEL=<Hugging Face Model to be converted to MaxText> # e.g. 'llama3.1-8b-Instruct'
+export HF_TOKEN=<Hugging Face access token> # your token to access gated HF repos
+
+# -- MaxText configuration --
+export MODEL_CHECKPOINT_DIRECTORY=<output directory to store output of checking point> # e.g., gs://my-bucket/my-checkpoint-directory
+
+# -- storage and format options
+USE_ZARR3=<Flag to use zarr3> # True to run SFT with McJAX, False to run SFT with Pathways
```
Comment (SurbhiJainUSC) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md]: nit: 
```
USE_ZARR3=<Flag to use zarr3> # Set to True to use zarr3 format (recommended for McJAX); set to False for Pathways.

USE_OCDBT=<Flag to use ocdbt> # Set to True to use OCDBT format (recommended for McJAX); set to False for Pathways.
```

File: docs/guides/checkpointing_solutions/convert_checkpoint.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    model_name=${HF_MODEL} \
+    hf_access_token=${HF_TOKEN} \
+    base_output_directory=${MODEL_CHECKPOINT_DIRECTORY} \
+    scan_layers=True \
+    use_multimodal=false \
+    hardware=cpu \
+    skip_jax_distributed_system=true \
+    checkpoint_storage_use_zarr3=${USE_ZARR3} \
+    checkpoint_storage_use_ocdbt=${USE_OCDBT} \
+    --lazy_load_tensors=true
```
Comment (SurbhiJainUSC) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md]: Move this to environment variable section

File: docs/guides/checkpointing_solutions/convert_checkpoint.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+**Key arguments:**
+
+  * `model_name`: The model identifier, which should be defined in `src/MaxText/utils/utils.py`.
+  * `scan_layers`: Indicates if the output checkpoint is [scanned](https://github.com/AI-Hypercomputer/maxtext/blob/main/docs/reference/core_concepts/checkpoints.md) (scan_layers=true) or unscanned (scan_layers=false).
+  * `use_multimodal`: Indicates if multimodality is used, important for Gemma3.
+  * `hf_access_token`: Your Hugging Face token.
+  * `base_output_directory`: The path where the converted Orbax checkpoint will be stored; it can be Googld Cloud Storage (GCS) or local. If not set, the default output directory is `Maxtext/tmp`.
+  * `hardware=cpu`: run the conversion script on a CPU machine.
+  * `checkpoint_storage_use_zarr3`: storage and format option, True to run SFT with McJAX, False to run SFT with Pathways
```
Comment (SurbhiJainUSC) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md]: same comment as above

File: docs/guides/checkpointing_solutions/convert_checkpoint.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  * `scan_layers`: Indicates if the output checkpoint is [scanned](https://github.com/AI-Hypercomputer/maxtext/blob/main/docs/reference/core_concepts/checkpoints.md)  (scan_layers=true) or unscanned (scan_layers=false).
+  * `hf_access_token`: Your Hugging Face token.
+  * `use_multimodal`: Indicates if multimodality is used, important for Gemma3.
+  * `base_output_directory`: The path where the converted Orbax checkpoint will be stored; it can be Googld Cloud Storage (GCS), Hugging Face Hub or local. If not set, the default output directory is `Maxtext/tmp`.
+  * `weight_dtype`: dtype for MaxText weights. It affects the resulting HF weight dtype. Default value is `float32`. We recommend using `bfloat16` to save memory and speed up conversion.
+
+
+## Verifying conversion correctness
+
+To ensure the conversion was successful, you can use the `tests/utils.forward_pass_logit_checker.py` script. It runs a forward pass on both the original and converted models and compares the output logits to verify conversion. It is used to verify the bidirectional conversion.
```
Comment (SurbhiJainUSC) [inline docs/guides/checkpointing_solutions/convert_checkpoint.md]: nit: `tests/utils/forward_pass_logit_checker.py`

Comment (charlesli640) [reply docs/guides/checkpointing_solutions/convert_checkpoint.md]: Fixed in commit 822fa8460

File: docs/guides/checkpointing_solutions/convert_checkpoint.md (hunk 4 above)
Comment (SurbhiJainUSC) [reply docs/guides/checkpointing_solutions/convert_checkpoint.md]: This is not fixed

File: docs/guides/checkpointing_solutions/convert_checkpoint.md (hunk 2 above)
Comment (SurbhiJainUSC) [reply docs/guides/checkpointing_solutions/convert_checkpoint.md]: This is not fixed
